tests/src/test/java/alluxio/client/fs/LocalCacheFileInStreamIntegrationTest.java

```diff
@@ -0,0 +1,206 @@
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.fs;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.AlluxioURI;
+import alluxio.Constants;
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.FileSystemContext;
+import alluxio.client.file.FileSystemTestUtils;
+import alluxio.client.file.URIStatus;
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.PageId;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.status.UnavailableException;
+import alluxio.grpc.WritePType;
+import alluxio.testutils.BaseIntegrationTest;
+import alluxio.testutils.LocalAlluxioClusterResource;
+import alluxio.util.io.BufferUtils;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.io.ByteStreams;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+
+public final class LocalCacheFileInStreamIntegrationTest extends BaseIntegrationTest {
+  private static final int PAGE_SIZE_BYTES = Constants.KB;
+  private static final int PAGE_COUNT = 32;
+  private static final int CACHE_SIZE_BYTES = PAGE_COUNT * PAGE_SIZE_BYTES;
+
+  @Rule
+  public LocalAlluxioClusterResource mClusterResource =
+      new LocalAlluxioClusterResource.Builder()
+          .setProperty(PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE, PAGE_SIZE_BYTES)
+          .setProperty(PropertyKey.USER_CLIENT_CACHE_SIZE, CACHE_SIZE_BYTES)
+          .setProperty(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT, Constants.MB)
+          .setProperty(PropertyKey.USER_LOCAL_CACHE_ENABLED, true)
+          .build();
+
+  @Rule
+  public final ExpectedException mThrown = ExpectedException.none();
+
+  private FileSystemContext mFsContext;
+  private CacheManager mCacheManager;
+  private FileSystem mFileSystem;
+  private String mFilePath;
+
+  @Before
+  public void before() throws Exception {
+    mFsContext = FileSystemContext.create(mClusterResource.get().getClient().getConf());
+    mCacheManager = mFsContext.getCacheManager();
+    mFileSystem = mClusterResource.get().getClient(mFsContext);
+    mFilePath = PathUtils.uniqPath();
+  }
+
+  @After
+  public void after() throws Exception {
+    mFsContext.close();
+  }
+
+  @Test
+  public void read() throws Exception {
```

Comment:
I actually suggest to remove the assertions around `mCacheManager` which is implementation details for integration tests (meant to be testing user-facing APIs and `mCacheManager` is not).

```java
 @Test
  public void read() throws Exception {
    AlluxioURI path = new AlluxioURI(mFilePath);
    FileSystemTestUtils.createByteFile(
        mFileSystem, mFilePath, WritePType.MUST_CACHE, PAGE_SIZE_BYTES);
    // read a file to populate the cache
    try (FileInStream stream = mFileSystem.openFile(path)) {
      assertTrue(BufferUtils.equalIncreasingByteArray(
          PAGE_SIZE_BYTES, ByteStreams.toByteArray(stream)));
    }
    mClusterResource.get().stopWorkers();
    // verify reading from local cache
    try (InputStream stream = mFileSystem.openFile(path)) {
      assertTrue(BufferUtils.equalIncreasingByteArray(
          PAGE_SIZE_BYTES, ByteStreams.toByteArray(stream)));
    }
  }
```

Reply:
I am fine removing it here since we test the cache indirectly through offline read. I wouldn't rule out using implementation details in integration tests where it is hard to verify results just through the public interface.